pkg/protocol/engine/mempool/v1/mempool.go

```diff
@@ -197,6 +197,42 @@ func (m *MemPool[VoteRank]) stateDiff(slot iotago.SlotIndex) (*StateDiff, error)
 	return lo.Return1(m.stateDiffs.GetOrCreate(slot, func() *StateDiff { return NewStateDiff(slot, kv) })), nil
 }
 
+// Reset resets the component to a clean state as if it was created at the last commitment.
+func (m *MemPool[VoteRank]) Reset() {
+	stateDiffsToDelete := make([]iotago.SlotIndex, 0)
```

Comment:
same

Reply:
Foreach grabs a read lock.

Reply:
I think we can clean the whole structure as it only contains data for uncommitted slots.

Reply:
I changed the code so everywhere where we can actually delete inline or clear, we do it.